library/ceph_key.py

```diff
@@ -0,0 +1,422 @@
+    cmd_list.append(generate_ceph_cmd(cluster, args, containerized))
+
+    return cmd_list
+
+
+def update_key(cluster, name, caps, containerized):
+    '''
+    Update a CephX key's capabilities
+    '''
+
+    cmd_list = []
+
+    args = [
+        "caps",
+        name,
+    ]
+
+    args = generate_caps(args, "ceph", caps)
+    cmd_list.append(generate_ceph_cmd(cluster, args, containerized))
+
+    return cmd_list
+
+
+def delete_key(cluster, name, containerized):
+    '''
+    Delete a CephX key
+    '''
+
+    cmd_list = []
+
+    args = [
+        "del",
+        name,
+    ]
+
+    cmd_list.append(generate_ceph_cmd(cluster, args, containerized))
+
+    return cmd_list
+
+
+def info_key(cluster, name, containerized):
+    '''
+    Get information about a CephX key
+    '''
+
+    cmd_list = []
+
+    args = [
+        "get",
+        name,
+        '-f',
+        'json',
+    ]
+
+    cmd_list.append(generate_ceph_cmd(cluster, args, containerized))
+
+    return cmd_list
+
+
+def list_keys(cluster, containerized):
+    '''
+    List all CephX keys
+    '''
+
+    cmd_list = []
+
+    args = [
+        "ls",
+        '-f',
+        'json',
+    ]
+
+    cmd_list.append(generate_ceph_cmd(cluster, args, containerized))
+
+    return cmd_list
+
+
+def exec_commands(module, cmd_list):
+    '''
+    Execute command(s)
+    '''
+
+    for cmd in cmd_list:
+        rc, out, err = module.run_command(cmd, encoding=None)
+        if rc != 0:
+            return rc, cmd, out, err
+
+    return rc, cmd, out, err
+
+
+def run_module():
+    module_args = dict(
+        cluster=dict(type='str', required=False, default='ceph'),
+        name=dict(type='str', required=False),
+        state=dict(type='str', required=True),
+        containerized=dict(type='str', required=True, default=None),
+        caps=dict(type='dict', required=False, default=None),
+        secret=dict(type='str', required=False, default=None),
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    # Gather module parameters in variables
+    cluster = module.params['cluster']
+    containerized = module.params['containerized']
+    name = module.params['name']
+    state = module.params['state']
+    caps = module.params['caps']
+    secret = module.params['secret']
+
+    result = dict(
+        changed=False,
+        stdout='',
+        stderr='',
+        rc='',
+        start='',
+        end='',
+        delta='',
+    )
+
+    if module.check_mode:
+        return result
+
+    startd = datetime.datetime.now()
+
+    if state == "present":
+        if caps is None:
+            fatal("Capabilities must be provided when state is 'present'", module)
+
+        if len(secret) == 0:
+            secret = None
+
+        # Test if the key exists, if it does we skip its creation
+        rc, cmd, out, err = exec_commands(module, info_key(cluster, name, containerized))
```

Comment:
this line should be moved before the `if` L349

Reply:
thanks